packages/cli/src/models/local/KitController.ts

```diff
@@ -0,0 +1,116 @@
+import path from 'path';
+import axios from 'axios';
+import fs from 'fs-extra';
+import Ajv from 'ajv';
+
+import stdout from '../../utils/stdout';
+import patch from '../../utils/patch';
+import child from '../../utils/child';
+import Spinner from '../../utils/spinner';
+
+const simpleGit = patch('simple-git/promise');
+
+export const DEFAULT_VERSION = '0.1.0';
+
+export interface KitConfig {
+  version: string;
+  message: string;
+  files: string[];
+  hooks: object;
+}
+
+/* tslint:disable */
+// schema has to have doulbe quotes
+const kitConfigSchema = {
+  "type": "object",
+  "$schema": "http://json-schema.org/draft-07/schema#",
+  "properties": {
+    "version": { "type": "string" },
+    "message": { "type": "string" },
+    "files": { "type": "array" },
+    "hooks": { "type": "object" },
+  },
+  "additionalProperties": false,
+  "required": ["version", "message", "files", "hooks"],
+};
+/* tslint:enable */
+
+export default class KitController {
+  public async unpack(url: string, workingDirPath: string = '', config: KitConfig): Promise<void | never> {
+    if (!url) throw Error('A url must be provided.');
+    if (!config) throw Error('A config must be provided.');
+
+    const { exec } = child;
+    const { readdir, remove, pathExists } = fs;
+
+    // because zos always spawns '.zos.lock' file
+    if ((await readdir(workingDirPath)).length > 1) throw Error('The directory must be empty');
```

Comment:
I'd suggest checking that the content is indeed `zos.lock`. If the user specifies a different working dir, then it should be absolutely empty, since `zos.lock` should not be there.